doc/7/controllers/document/upsert/index.md

```diff
@@ -0,0 +1,44 @@
+---
+code: true
+type: page
+title: upsert
+---
+
+# upsert
+
+<SinceBadge version="2.8.0"/>
+
+Applies partial changes to a document. If the document doesn't already exist, a new document is created.
+
+
+```js
+update(index, collection, id, document, [options]);
+```
+
+| Argument     | Type              | Description                               |
+| ------------ | ----------------- | ----------------------------------------- |
+| `index`      | <pre>string</pre> | Index name                                |
+| `collection` | <pre>string</pre> | Collection name                           |
+| `id`         | <pre>string</pre> | Document ID                               |
+| `changes`    | <pre>object</pre> | Partial content of the document to update |
+| `defaults`   | <pre>object</pre><br/>(`{}`) | Fields to add to the document if it gets created 
+
+
+### Options
+
+Additional query options
+
+| Options           | Type<br/>(default)              | Description                                                                        |
+| ----------------- | ------------------------------- | ---------------------------------------------------------------------------------- |
+| `refresh`         | <pre>string</pre><br/>(`""`)    | If set to `wait_for`, waits for the change to be reflected for `search` (up to 1s) |
+| `retryOnConflict` | <pre>int</pre><br/>(`0`)        | The number of times the database layer should retry in case of version conflict    |
+| `source`          | <pre>boolean</pre><br/>(`false`)| If true, returns the updated document inside the response
+
+
+## Resolves
+
+Resolves to an object containing the document update result.
```

Comment:
I'm not sure I understand
According to the documentation it always returns the `_id` and the `version` of the document, and in the js sdk those 2 belong to the `Document` class. The `_source` is an `DoucmentContentClass` embedded in it, which means that it would stay `null` if the option is not provided.
So I'm not sure to see the problem here? Wdyt?

Reply:
IMHO you can't return the document type here because it is also missing the `created` property.

Reply:
I created it